lib/Models/Catalog/Esri/ArcGisMapServerCatalogItem.ts

```diff
@@ -153,10 +153,17 @@ class MapServerStratum extends LoadableStratum(
 
       if (isDefined(layersMetadataResponse?.layers)) {
         layers = layersMetadataResponse.layers;
-        // If layersMetadata is only a single layer -> shove into an array
-      } else if (isDefined(layersMetadataResponse?.id)) {
-        layers = [layersMetadataResponse];
       }
+      // The following conditional code is commented out becasue
+      // the test will be true when the requested url path ends with
+      // layer ID. Overwriting "layers" will make "stratum.allLayers"
+      // contain single layer instead of all layers that were set
+      // previously from the base URI, causing unexpected errors.
+      //
+      // If layersMetadata is only a single layer -> shove into an array
+      // else if (isDefined(layersMetadataResponse?.id)) {
+      // layers = [layersMetadataResponse];
+      // }
```

Comment:
It might be better to remove the case where layers metadata is returned for a single layer.

This could be done by changing

https://github.com/TerriaJS/terriajs/blob/48f705486fde742dd8a9fcc743890b2179f7c590/lib/Models/Catalog/Esri/ArcGisMapServerCatalogItem.ts#L113

to just

```ts
let layersUri = getBaseURI(item).segment("layers");
```

Then you can remove the entire block highlighted by this comment. 

This would mean that we get "detailed" layer metadata for all layers regardless of if layers have been specified (in the `url` or the `layers` trait) - which should fix the bug you found where `allLayers` only contained speciried layers

Reply:
I am not entirely sure of the difference between "basic" and "detailed" layer metadata.

The "basic" layer metadata is the `layers` object in the MapServer service metadata. The "detailed" layer metadata is fetched by calling the `layers` MapServer endpoint.

https://github.com/TerriaJS/terriajs/blob/48f705486fde742dd8a9fcc743890b2179f7c590/lib/Models/Catalog/Esri/ArcGisMapServerCatalogItem.ts#L149-L156